stitchee/concatenate.py

```diff
@@ -139,6 +140,104 @@ def validate_concat_method_and_dim(concat_method: str, concat_dim: str | None =
         )
 
 
+def find_variables_with_duplicate_dimensions(dataset: xr.DataSet):
+    # find variables with duplicate dimensions
+    # by comparing dimension's index with its first occurrence
+    duplicate_variables = [
+        varname
+        for varname in dataset.variables
+        if (
+            (dims := dataset[varname].dims)
+            and (any(idim != dims.index(dim) for idim, dim in enumerate(dims)))
+        )
+    ]
+
+    return duplicate_variables
+
+
+def rename_to_uniq_dimensions(dataarray):
+    # find the repetition numbers for all dimensions
```

Comment:
I think this could use a docstring:

```suggestion
def rename_to_uniq_dimensions(dataarray: xr.DataArray):
    """Make duplicate dimension names unique by adding numeric prefixes.
        
    Returns
    -------
    xr.DataArray
        New DataArray with unique dimension names and updated coordinates.
        
    Examples
    --------
    >>> arr = xr.DataArray([[1, 2], [3, 4]], dims=['x', 'x'])
    >>> result = rename_to_uniq_dimensions(arr)
    >>> result.dims
    ('x', '1___x')
    """
```